contrib/guix/manifest.scm

```diff
@@ -520,6 +527,12 @@ inspecting signatures in Mach-O binaries.")
 (define (make-glibc-without-werror glibc)
   (package-with-extra-configure-variable glibc "enable_werror" "no"))
 
+(define (make-glibc-with-stack-protector glibc)
+  (package-with-extra-configure-variable glibc "--enable-stack-protector" "all"))
```

Comment:
glibs docs [suggest](https://sourceware.org/legacy-ml/libc-alpha/2017-02/msg00079.html):
> It is recommended to build glibc with `--enable-stack-protector=strong`.

Why has `--enable-stack-protector=all` been suggested?

Reply:
> Why has --enable-stack-protector=all been suggested?

because I'm turning it on for all functions.

Reply:
> > Why has --enable-stack-protector=all been suggested?
> 
> because I'm turning it on for all functions.

Yes, I see this. But my question was about not following glibs docs recommendations. I assume glibc authors have reasons for them. As you suggest a different option value, why do you think it is better than the recommended one?

Reply:
I choose `all` because I want to instrument all functions, rather than leaving it up to compiler heuristics/params (ssp-buffer-size) to decide which functions to instrument or not.